File: objc/Project16/Project16/StarFormatter.m
//
//  StarFormatter.m
//  Project16
//
//  Created by Jinwoo Kim on 1/13/21.
//

#import "StarFormatter.h"

@implementation StarFormatter
- (NSString *)stringForObjectValue:(id)obj {
    if (obj) {
        NSInteger number = [[NSString stringWithFormat:@"%@", obj] integerValue];
        if (number) {
            NSString *result = @"";
            for (NSInteger i = 0; i < number; i++) {
                result = [NSString stringWithFormat:@"%@⭐️", result];
            }
            return result;
        }
    }
    return @"";
}
@end

File: swift/Project16/Project16/StarFormatter.swift
//
//  StarFormatter.swift
//  Project16
//
//  Created by Jinwoo Kim on 1/13/21.
//

import Cocoa

class StarFormatter: Formatter {
    override func string(for obj: Any?) -> String? {
        if let obj = obj {
            if let number = Int(String(describing: obj)) {
                return String(repeating: "⭐️", count: number)
            }
        }
        return ""
    }
}
